<launch>
    
    <!-- Spawn Task-2 Models in Gazebo -->
    <include file="$(find pkg_vb_sim)/launch/task5_world.launch" />

    <!-- Spawn two UR5 in Gazebo -->
    <include file = "$(find pkg_vb_sim)/launch/two_ur5_gazebo.launch" />

    <!-- Spawn T5 Boxes -->
    <rosparam file ="$(find pkg_vb_sim)/config/config_package_colour.yaml"/>
    <node name= "task5_spawn_models" pkg= "pkg_vb_sim" type="task5_spawn_models.py"/>

    <!-- Run Conveyor Belt Service Server -->
    <node name= "node_service_server_conveyor_belt" pkg= "pkg_vb_sim" type="node_service_server_conveyor_belt.py" output="screen"/>

    <!-- Run the Online Order Placer Node -->
    <rosparam file ="$(find pkg_vb_sim)/config/config_online_order.yaml"/>
    <node name= "node_online_order_placer" pkg= "pkg_vb_sim" type="node_online_order_placer.py" output="screen"/>
    
    <rosparam file ="$(find pkg_task5)/config/config_pyiot.yaml"/>

    <!-- for UR5_1 -->
    <group ns="ur5_1">
        <!-- <rosparam command="load" file="$(find pkg_moveit_ur5)/config/joint_names.yaml" /> -->

        <include file="$(find pkg_moveit_ur5_1_config)/launch/planning_context.launch">
          <arg name="load_robot_description" value="true"/>
        </include>

        <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
          <param name="/use_gui" value="false"/>
          <rosparam param="/source_list">[/joint_states]</rosparam>
        </node>

        <!-- Launch Moveit move_group Node  -->
        <include file="$(find pkg_moveit_ur5_1_config)/launch/move_group.launch">
          <arg name="publish_monitored_planning_scene" value="true"/>
        </include>

        <include file="$(find pkg_moveit_ur5_1_config)/launch/moveit_rviz_robot1.launch">
          <arg name="config" value="true"/>
        </include>

       <!-- Adding shelf and 2D camera to moveit! Planning scene  in Rviz -->
        <!-- <arg name="scene_file" default="$(find pkg_task4)/config/rviz/ur5_1_with_shelf.scene" /> -->
         <!--   <node name = "moveit_publish_scene_from_text" pkg="moveit_ros_planning" type="moveit_publish_scene_from_text" args="$(arg scene_file)" /> -->

    </group>



   <!-- for UR5_2  -->
    <group ns="ur5_2">
        <!-- <rosparam command="load" file="$(find pkg_moveit_ur5)/config/joint_names.yaml" /> -->

        <include file="$(find pkg_moveit_ur5_2_config)/launch/planning_context.launch">
          <arg name="load_robot_description" value="true"/>
        </include>

        <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
          <param name="/use_gui" value="false"/>
          <rosparam param="/source_list">[/joint_states]</rosparam>
        </node>
        
        <!-- Launch Moveit move_group Node  -->
        <include file="$(find pkg_moveit_ur5_2_config)/launch/move_group.launch">
          <arg name="publish_monitored_planning_scene" value="true"/>
        </include>

        <include file="$(find pkg_moveit_ur5_2_config)/launch/moveit_rviz_robot2.launch">
          <arg name="config" value="true"/>
        </include>

        <!-- Adding bins and conveyor belt to moveit! Planning scene  in Rviz -->
        <!-- <arg name="scene_file" default="$(find pkg_task4)/config/rviz/ur5_2_sorting_bin.scene" /> -->
           <!--  <node name = "moveit_publish_scene_from_text" pkg="moveit_ros_planning" type="moveit_publish_scene_from_text" args="$(arg scene_file)" /> -->
    </group>

	
</launch>